src/internal/minikubetestenv/deploy.go

```diff
@@ -878,13 +880,50 @@ func putRelease(t testing.TB, ctx context.Context, namespace string, kubeClient
 	previousOptsHash := getOptsConfigMapData(t, ctx, kubeClient, namespace)
 	pachdExists, err := checkForPachd(t, ctx, kubeClient, namespace, opts.EnterpriseServer)
 	require.NoError(t, err)
-	if mustUpgrade {
+	if mustUpgrade || opts.InstallPrometheus {
 		t.Logf("Test must upgrade cluster in place, upgrading cluster with new opts in %v", namespace)
 		require.NoErrorWithinTRetry(t,
 			time.Minute,
 			func() error {
 				return errors.EnsureStack(helm.UpgradeE(t, helmOpts, chartPath, namespace))
```

Comment:
Why does opts.InstallPrometheus trigger a helm upgrade of the pachyderm chart?

Reply:
You’re right — it doesn’t really make sense (it was originally a path dependency). I’ve split it out.